haystack/pipelines/base.py

```diff
@@ -865,12 +969,23 @@ def save_to_yaml(self, path: Path, return_defaults: bool = False):
         Save a YAML configuration for the Pipeline that can be used with `Pipeline.load_from_yaml()`.
 
         :param path: path of the output YAML file.
+        :param return_defaults: whether to output parameters that have the default values.
+        """
+        config = self.get_config(return_defaults=return_defaults)
+        with open(path, "w") as outfile:
+            yaml.dump(config, outfile, default_flow_style=False)
+
+    def get_config(self, return_defaults: bool = False) -> dict:
+        """
+        Returns a configuration for the Pipeline that can be used with `Pipeline.load_from_config()`.
+
         :param return_defaults: whether to output parameters that have the default values.
```

Comment:
Do you think it's necessary? I can't see a case where you don't want to return defaults. Doesn't hurt to leave it in though, I guess :slightly_smiling_face: